### ISSUE TITLE
Auto update submodules

### ISSUE THREAD

**Issue by abelgladstone** [Microsoft/vscode#56249]
Issue Type: <b>Bug</b>

Git submodule information (branch names, modified files) do not auto update in the source control view. 
For example: If you are working on a branch and switched to a new branch in both the main repo and a submodule. The view still shows branch information that is from the old branch. The same can be observed when you commit changes, The source control view is not updated.

You can fix this by clicking on the refresh button.

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:43:53.668Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 x 2693)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.62GB (22.39GB free)|
|Process Argv|c:\program files\microsoft vs code\code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eclipse-keybindings|alp|0.9.8
file-templates|brp|1.2.0
gitlens|eam|8.5.4
vs-code-xml-format|fab|0.1.5
gc-excelviewer|Gra|2.1.25
python|ms-|2018.7.1
cpptools|ms-|0.17.7
material-icon-theme|PKi|3.5.2
ruby|reb|0.20.0
java|red|0.29.0
cmake|twx|0.0.17
shell-launcher|Tyr|0.2.0
cmake-tools|vec|1.1.1

(1 theme extensions excluded)

</details>

**Comment by joaomoreno** [comment]
Dependent on #3025, duplicate of #42821
